hydromt/driver/zarr_driver.py

```diff
@@ -0,0 +1,63 @@
+"""RasterDatasetDriver for zarr data."""
+
+from functools import partial
+from logging import Logger
+from typing import Callable, List, Optional
+
+import xarray as xr
+from pyproj import CRS
+
+from hydromt._typing import Bbox, Geom
+from hydromt._typing.error import NoDataStrategy
+from hydromt.driver.preprocessing import PREPROCESSORS
+from hydromt.driver.rasterdataset_driver import RasterDatasetDriver
+
+
+class ZarrDriver(RasterDatasetDriver):
+    """RasterDatasetDriver for zarr data."""
+
+    name = "zarr"
+
+    def read(
+        self,
+        uri: str,
+        *,
+        bbox: Optional[Bbox] = None,
+        mask: Optional[Geom] = None,
+        buffer: float = 0,
+        crs: Optional[CRS] = None,
+        variables: Optional[List[str]] = None,
+        predicate: str = "intersects",
+        zoom_level: int = 0,
+        logger: Optional[Logger] = None,
+        handle_nodata: NoDataStrategy = NoDataStrategy.RAISE,
+        # TODO: https://github.com/Deltares/hydromt/issues/802
+        **kwargs,
+    ) -> xr.Dataset:
+        """
+        Read zarr data to an xarray DataSet.
+
+        Args:
+        """
+        uris = self.metadata_resolver.resolve(
+            uri,
+            bbox=bbox,
+            geom=mask,
+            buffer=buffer,
+            predicate=predicate,
+            variables=variables,
+            handle_nodata=handle_nodata,
+            **kwargs,
+        )
+        preprocessor: Optional[Callable] = None
+        preprocessor_name: Optional[str] = kwargs.get("preprocess")
+        if preprocessor_name:
+            preprocessor = PREPROCESSORS.get(preprocessor_name)
+            if not preprocessor:
+                raise ValueError(f"unknown preprocessor: '{preprocessor_name}'")
```

Comment:
You were doing all these nice walrus operators. Would something like this work too?

```suggestion
        if preprocessor_name and not preprocessor := PREPROCESSORS.get(preprocessor_name):
                raise ValueError(f"unknown preprocessor: '{preprocessor_name}'")
```

Reply:
The variable declaration has to be after the `if` keyword unfortunately.